src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -255,16 +256,34 @@ tx_gateway_frontend::find_coordinator(kafka::transactional_id id) {
             co_return std::nullopt;
         }
     }
+    auto tx_ntp_opt = get_ntp(id);
+    if (!tx_ntp_opt) {
+        co_return std::nullopt;
+    }
+    auto tx_ntp = tx_ntp_opt.value();
+    auto x = _metadata_cache.local().get_leader_id(tx_ntp);
+    co_return x;
+}
 
-    co_return _metadata_cache.local().get_leader_id(model::legacy_tm_ntp);
+std::optional<model::ntp>
+tx_gateway_frontend::get_ntp(kafka::transactional_id id) {
+    auto tx_ntp_opt = _tx_coordinator_ntp_mapper.local().ntp_for(
+      kafka::group_id(id));
+    if (!tx_ntp_opt) {
+        vlog(
+          txlog.debug,
+          "Topic {} doesn't exist in metadata cache",
+          model::tx_manager_nt);
```

Comment:
does this message need to be updated? it's neither using the metadata cache nor is the query for a topic it is for a group-id (though i suspect that is implied since it's probably a modulo mapping to however many partitions do exist?)

Reply:
Only situation when we get nullopt here is when topic doesn't exist in metadata cache

```
    std::optional<model::ntp> ntp_for(const kafka::group_id& group) const {
        auto cfg = _md.local().get_topic_cfg(_tp_ns);
        if (!cfg) {
            return std::nullopt;
        }
        incremental_xxhash64 inc;
        inc.update(group);
        auto p = static_cast<model::partition_id::type>(
          jump_consistent_hash(inc.digest(), cfg->partition_count));
        return model::ntp(_tp_ns.ns, _tp_ns.tp, model::partition_id{p});
    }
```